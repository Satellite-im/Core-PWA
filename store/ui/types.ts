import { TranslateResult } from 'vue-i18n'
import { FileMessage } from '~/types/textile/mailbox'
import { Glyph } from '~/types/ui/glyph'
import { Channel } from '~/types/ui/server'
import { Alert } from '~/libraries/ui/Alerts'
import { ConversationMessage } from '~/libraries/Iridium/chat/types'

export enum GlyphMarketViewStatus {
  HOME = 'home',
  SHOP_ALL = 'shop_all',
  SHOP_DETAIL = 'shop_detail',
}

export enum ModalWindows {
  NEW_FOLDER = 'newfolder',
  CREATE_SERVER = 'createServer',
  MARKETPLACE = 'marketplace',
  WALLET = 'wallet',
  WALLET_MINI = 'walletMini',
  ERROR = 'error',
  CHANGELOG = 'changelog',
  GLYPH = 'glyph',
  USER_PROFILE = 'userProfile',
  CALL_TO_ACTION = 'callToAction',
  RENAME_FILE = 'renameFile',
}

export interface EnhancerInfo {
  show: boolean
  floating?: boolean
  position?: number[]
  defaultWidth?: string
  defaultHeight?: string
  containerWidth?: number
  route: string
}

export interface EmojiUsage {
  code: string
  count: number
  content: string
}

export interface RecentGlyph {
  pack: Glyph
  url: string
  count: number
}

export enum SettingsRoutes {
  PERSONALIZE = 'personalize',
  PROFILE = 'profile',
  AUDIO_AND_VIDEO = 'audio & Video',
  KEY_BINDS = 'keybinds',
  ACCOUNTS_AND_DEVICES = 'accounts & Devices',
  PRIVACY = 'privacy',
  DEVELOPER = 'developer',
  INFO = 'info',
  NOTIFICATIONS = 'notifications',
  STORAGE = 'storage',
  NETWORK = 'network',
  REALMS = 'realms',
}

export type ContextMenuItemTypes = 'primary' | 'danger'

export interface ContextMenuItem {
  text: string | TranslateResult
  func: Function
  type?: ContextMenuItemTypes
}

export interface UIState {
  contextMenuStatus: boolean
  contextMenuValues: ContextMenuItem[]
  quickProfile: object | boolean
  userProfile: object
  notifications: Alert[]
  contextMenuPosition: object
  quickProfilePosition: object
  showSettings: boolean
  settingsSideBar: boolean
  settingsRoute: SettingsRoutes
  showSidebarUsers: boolean
  showSearchResult: boolean
  showSidebar: boolean
  modals: {
    [key in ModalWindows]: boolean | object
  }
  glyphModalPack: string
  chatbarContent: string
  chatbarFocus: boolean
  replyChatbarMessage?: ConversationMessage
  showPinned: boolean
  fullscreen: boolean
  enhancers: EnhancerInfo
  messages: any[]
  unreadMessage: number
  isScrollOver: boolean
  showOlderMessagesInfo: boolean
  isTyping: object | boolean
  isReacted: boolean
  activeChannel: Channel | undefined
  settingReaction: object
  hoveredGlyphInfo: object | undefined
  glyphMarketplaceView: object
  editMessage: {
    id: string
    from: string
    payload: string
  }
  mostEmojiUsed: EmojiUsage[]
  recentGlyphs: RecentGlyph[]
  chatImageOverlay?: FileMessage
  isMobileNavVisible: boolean
}

export type Position = {
  x: number
  y: number
}
