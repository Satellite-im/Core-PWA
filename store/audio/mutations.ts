import { AudioState } from './types'

const mutations = {
  mute(state: AudioState) {
    state.muted = !state.muted
  },
  toggleMute(state: AudioState) {
    state.muted = !state.muted
  },
  deafen(state: AudioState) {
    const isDeafened = state.deafened
    state.previousVolume = isDeafened ? state.previousVolume : state.volume
    state.volume = isDeafened ? state.previousVolume : 0
    state.deafened = !isDeafened
  },
  setVolume(state: AudioState, volume: Number) {
    state.previousVolume = volume
    state.volume = volume
  },
  setSystemVolume(state: AudioState, systemVolume: Number) {
    state.sounds.system = systemVolume
  },
  setInputVolume(state: AudioState, inputVolume: Number) {
    state.inputVolume = inputVolume
  },
  setSoundLevel(
    state: AudioState,
    sound: {
      inboundMedia: Number
      outboundMedia: Number
      system: Number
    },
  ) {
    state.sounds = sound
  },
}

export default mutations
