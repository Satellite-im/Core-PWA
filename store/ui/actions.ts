import Mousetrap from 'mousetrap'
import Vue from 'vue'
import { v4 as uuidv4 } from 'uuid'
import { Position, UIState } from './types'
import SoundManager, { Sounds } from '~/libraries/SoundManager/SoundManager'
import TextileManager from '~/libraries/Textile/TextileManager'
import { ActionsArguments } from '~/types/store/store'
import { Friend } from '~/types/ui/friends'
import { Channel } from '~/types/ui/server'
import { getFullUserInfoFromState } from '~/utilities/Messaging'
import { getCorrectKeybind } from '~/utilities/Keybinds'
import { TextileError } from '~/store/textile/types'
import { AlertType } from '~/libraries/ui/Alerts'

const $Sounds = new SoundManager()

export default {
  setMessages({ commit }: ActionsArguments<UIState>, messages: any[]) {
    commit('setMessages', messages)
  },
  sendMessage({ commit, rootState }: ActionsArguments<UIState>, message: any) {
    if (message.user.address !== rootState.accounts.active) {
      $Sounds.playSound(Sounds.NEW_MESSAGE)
    }
    commit('sendMessage', message)
  },
  setIsScrollOver({ commit }: ActionsArguments<UIState>, status: boolean) {
    commit('setIsScrollOver', status)
  },
  setIsReacted({ commit }: ActionsArguments<UIState>, status: boolean) {
    commit('setIsReacted', status)
  },
  setActiveChannel({ commit }: ActionsArguments<UIState>, channel: Channel) {
    commit('setActiveChannel', channel)
  },
  /**
   * @method openSettings
   * @description Opens setting page
   * @example Mousetrap.bind('ctrl+s', dispatch('audio/toggleMute') )
   */
  openSettings({ commit, state }: any) {
    commit('toggleSettings', { show: !state.showSettings })
  },
  /**
   * @method activateKeybinds
   * @description Activates all keybindings with Mousetrap
   * @example mounted (){ activateKeybinds() }
   */
  async activateKeybinds({ dispatch, rootState }: ActionsArguments<UIState>) {
    const { toggleMute, toggleDeafen, openSettings, callActiveChat } =
      // @ts-ignore
      rootState.settings.keybinds
    Mousetrap.reset()
    Mousetrap.bind(getCorrectKeybind(toggleMute), (event: KeyboardEvent) => {
      event.preventDefault()
      dispatch('audio/toggleMute', null, { root: true })
    })
    Mousetrap.bind(getCorrectKeybind(toggleDeafen), (event: KeyboardEvent) => {
      event.preventDefault()
      dispatch('audio/toggleDeafen', null, { root: true })
    })
    Mousetrap.bind(getCorrectKeybind(openSettings), (event: KeyboardEvent) => {
      event.preventDefault()
      dispatch('openSettings')
    })
    Mousetrap.bind(
      getCorrectKeybind(callActiveChat),
      (event: KeyboardEvent) => {
        event.preventDefault()
        dispatch('webrtc/call', { kinds: ['audio'] }, { root: true })
      },
    )
  },
  /**
   * @method setNotifications
   * @description Collects all existing notifications for a user
   */
  async setNotifications({ commit }: ActionsArguments<UIState>) {
    const $TextileManager: TextileManager = Vue.prototype.$TextileManager

    if (!$TextileManager.notificationManager?.isInitialized()) {
      throw new Error(TextileError.MAILBOX_MANAGER_NOT_INITIALIZED)
    }
    const notifications =
      await $TextileManager.notificationManager?.getnotifications()
    commit('setNotifications', notifications)
  },
  async sendNotification(
    { commit, rootState }: ActionsArguments<UIState>,
    payload: {
      message: string
      from: string
      imageHash: string
      title: string
      type: AlertType
    },
  ) {
    const $TextileManager: TextileManager = Vue.prototype.$TextileManager

    if (!$TextileManager.notificationManager?.isInitialized()) {
      throw new Error(TextileError.MAILBOX_MANAGER_NOT_INITIALIZED)
    }
    const notificationResponse =
      await $TextileManager.notificationManager?.sendNotification({
        from: payload.from,
        id: uuidv4(),
        title: payload.title,
        imageHash: payload.imageHash,
        message: payload.message,
        type: payload.type,
      })
    commit('sendNotification', notificationResponse)
  },
  removeSeenNotification(
    { commit, rootState }: ActionsArguments<UIState>,
    notificationId: string,
  ) {
    commit('notificationSeen', notificationId)
  },
  /**
   * @method clearKeybinds
   * @description Unbinds all current keybindings with Mousetrap
   * @example destroyed (){ clearKeybinds() }
   */
  async clearKeybinds({ dispatch }: ActionsArguments<UIState>) {
    Mousetrap.reset()
  },
  async setChatbarFocus({ dispatch }: ActionsArguments<UIState>) {
    await dispatch('toggleChatbarFocus', false)
    dispatch('toggleChatbarFocus', true)
  },
  toggleChatbarFocus({ commit }: ActionsArguments<UIState>, flag: boolean) {
    commit('setChatbarFocus', flag)
  },
  setChatbarContent(
    { commit, dispatch }: ActionsArguments<UIState>,
    val: {
      content: string
      userId?: string
    },
  ) {
    commit('chatbarContent', val.content)
    if (val.userId)
      dispatch(
        'chat/setChatText',
        { value: val.content, userId: val.userId },
        { root: true },
      )
  },
  showQuickProfile(
    { commit, state, rootState }: ActionsArguments<UIState>,
    payload: { textilePublicKey: string; position: Position },
  ) {
    if (!payload?.textilePublicKey || !payload?.position) {
      return
    }

    const selectedUser = getFullUserInfoFromState(
      payload.textilePublicKey,
      rootState,
    )

    commit('setQuickProfilePosition', payload.position)
    commit('quickProfile', selectedUser)
  },
  async showProfile(
    { commit, rootState, dispatch }: ActionsArguments<UIState>,
    user: Friend,
  ) {
    if (!user) {
      return
    }
    let metadata = null
    if (!user.metadata) {
      const $TextileManager: TextileManager = Vue.prototype.$TextileManager
      metadata = await $TextileManager.metadataManager?.getFriendMetadata(
        user.address,
      )
    }
    commit('toggleModal', {
      name: 'userProfile',
      state: true,
    })
    const friend = {
      ...user,
    }
    if (metadata) {
      friend.metadata = metadata
      commit('friends/updateFriend', friend, { root: true })
    }
    commit('setUserProfile', friend)
  },
}
