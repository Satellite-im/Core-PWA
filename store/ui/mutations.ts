import { NuxtState } from '@nuxt/types/app'

export default {
  toggleContextMenu(state: NuxtState, enabled: Boolean) {
    state.ui = {
      ...state.ui,
      contextMenuStatus: enabled,
    }
  },
  showSidebarUsers(state: NuxtState, enabled: Boolean) {
    state.ui = {
      ...state.ui,
      showSidebarUsers: enabled,
    }
  },
  setContextMenuValues(state: NuxtState, values: any) {
    state.ui = {
      ...state.ui,
      contextMenuValues: values,
    }
  },
  setContextMenuPosition(state: NuxtState, e: any) {
    state.ui = {
      ...state.ui,
      contextMenuPosition: { x: e.x, y: e.y },
    }
  },
  setQuickProfilePosition(state: NuxtState, e: any) {
    state.ui = {
      ...state.ui,
      quickProfilePosition: { x: e.x, y: e.y },
    }
  },
  quickProfile(state: NuxtState, profile: Object | Boolean) {
    state.ui = {
      ...state.ui,
      quickProfile: profile,
    }
  },
  chatbarContent(state: NuxtState, content: String) {
    state.ui = {
      ...state.ui,
      chatbarContent: content,
    }
  },
  fullscreen(state: NuxtState, fullscreen: Boolean) {
    state.ui = {
      ...state.ui,
      fullscreen,
    }
  },
  toggleEnhancers(state: NuxtState, show: Boolean) {
    state.ui = {
      ...state.ui,
      showEnhancers: show,
    }
  },
}
