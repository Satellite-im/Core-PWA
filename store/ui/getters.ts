import { EmojiUsage, UIState } from './types'
import { Alert, AlertState } from '~/libraries/ui/Alerts'

const getters = {
  getSortedMostUsedEmojis: (state: UIState): EmojiUsage[] => {
    return [...state.mostEmojiUsed].sort((a, b) => b.count - a.count)
  },
  getSortedRecentGlyphs: (state: UIState) => {
    return [...state.recentGlyphs].sort((a, b) => b.count - a.count)
  },
  showSidebar: (state: UIState) => {
    return state.showSidebar
  },
  allUnseenNotifications: (state: UIState): Alert[] => {
    return [...state.notifications]
      .sort((a, b) => {
        return b.at - a.at
      })
      .filter((noti) => {
        return noti.state === AlertState.UNREAD
      })
  },
  getAlertCount: (state: UIState) => {
    return state.notificationsAlertCounter
  },
  swiperSlideIndex: (state: UIState) => {
    return state.swiperSlideIndex
  },
  isFilesIndexLoading: (state: UIState): boolean => {
    return Boolean(state.filesUploadStatus)
  },
}

export default getters
