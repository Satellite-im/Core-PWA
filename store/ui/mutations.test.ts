import { User } from '~/libraries/Iridium/users/types'
import mutations from '~/store/ui/mutations'
import InitialUIState from '~/store/ui/state'
import { SettingsRoutes } from '~/store/ui/types'

// So we don't have annoying snapshot fails. (https://stackoverflow.com/questions/42935903/jest-snapshot-testing-how-to-ignore-part-of-the-snapshot-file-in-jest-test-resu)
Date.now = jest.fn(() => 1645617999076)
const dateNow = Date.now()

describe('mutations', () => {
  test('toggleContextMenu', () => {
    const localizedState = { ...InitialUIState() }
    mutations.toggleContextMenu(localizedState, true)
    expect(localizedState.contextMenuStatus).toBeTruthy()
  })

  test('showSidebar', () => {
    const localizedState = { ...InitialUIState() }
    mutations.showSidebar(localizedState, true)
    expect(localizedState.showSidebar).toBeTruthy()
  })

  test('setContextMenuValues', () => {
    const localizedState = { ...InitialUIState() }
    mutations.setContextMenuValues(localizedState, true)
    expect(localizedState.contextMenuValues).toBeTruthy()
  })

  test('setContextMenuPosition', () => {
    const localizedState = { ...InitialUIState() }
    const object = {
      x: 4,
      y: 2,
    }
    mutations.setContextMenuPosition(localizedState, object)
    expect(localizedState.contextMenuPosition).toMatchObject(object)
  })

  test('setFullProfile', () => {
    const localizedState = { ...InitialUIState() }
    const object: User = {
      did: '123',
      name: 'John',
    }
    mutations.setFullProfile(localizedState, object)
    expect(localizedState.fullProfile).toMatchObject(object)
  })

  test('chatbarContent', () => {
    const localizedState = { ...InitialUIState() }
    mutations.chatbarContent(localizedState, 'string')
    expect(localizedState.chatbarContent).toBe('string')
  })

  test('setChatImageOverlay', () => {
    const passedInImageOverlay = undefined
    const localizedState = { ...InitialUIState() }
    mutations.setChatImageOverlay(localizedState, passedInImageOverlay)
    expect(localizedState.chatImageOverlay).toBeUndefined()
  })

  test('setChatbarFocus', () => {
    const localizedState = { ...InitialUIState() }
    mutations.setChatbarFocus(localizedState, true)
    expect(localizedState.chatbarFocus).toBeTruthy()
  })

  test('setIsMobileNavVisible', () => {
    const localizedState = { ...InitialUIState() }
    mutations.setIsMobileNavVisible(localizedState, true)
    expect(localizedState.isMobileNavVisible).toBeTruthy()
  })

  test('setSettingsRoute', () => {
    const localizedState = { ...InitialUIState() }
    mutations.setSettingsRoute(localizedState, SettingsRoutes.PROFILE)
    expect(localizedState.settingsRoute).toBe(SettingsRoutes.PROFILE)
  })

  test('setGlyphModalPackId', () => {
    const localizedState = { ...InitialUIState() }
    mutations.setGlyphModalPackId(localizedState, 'string')
    expect(localizedState.glyphModalPackId).toBe('string')
  })

  test('toggleModal', () => {
    const localizedState = { ...InitialUIState() }
    const object = {
      name: 'quickchat',
      state: true,
    }
    mutations.toggleModal(localizedState, object)
    expect(localizedState.modals).toMatchObject({
      [object.name]: object.state,
    })

    /*
     * The reason we're doing the matcher the above way is rather than .toMatchObject(object) if we don't
     * it will be expecting:
     * {"name": "quickchat",
     * "state": true, }
     * rather than
     * {"quickchat": true}
     */
  })

  test('toggleModal with a non-existent modal', () => {
    const localizedState = { ...InitialUIState() }
    const object = {
      name: 'non-existent modal',
      state: true,
    }
    mutations.toggleModal(localizedState, object)
    expect(localizedState.modals).toMatchObject({
      [object.name]: object.state,
    })

    // This would not throw an error, because it will just create a new key with the boolean as a value
  })

  test('toggleErrorNetworkModal', () => {
    const localizedState = { ...InitialUIState() }
    const argument = { state: true, action: null }
    mutations.toggleErrorNetworkModal(localizedState, argument)
    expect(localizedState.modals.errorNetwork).toMatchObject({
      isOpen: argument.state,
      action: argument.action,
    })
  })

  test('setGlyphMarketplaceView', () => {
    const localizedState = { ...InitialUIState() }
    const object = { a: 'b' }
    mutations.setGlyphMarketplaceView(localizedState, object)
    expect(localizedState.glyphMarketplaceView).toBe(object)
  })

  test('setHoveredGlyphInfo', () => {
    const localizedState = { ...InitialUIState() }
    const object = { a: 'b' }
    mutations.setHoveredGlyphInfo(localizedState, object)
    expect(localizedState.hoveredGlyphInfo).toBe(object)
  })

  test('setEditMessage', () => {
    const localizedState = { ...InitialUIState() }
    const object = {
      payload: 'payload',
      id: '5d802d44-23c3-49d8-a725-407bd17eb56b',
      from: 'Retha Larkin',
    }
    mutations.setEditMessage(localizedState, object)
    expect(localizedState.editMessage).toMatchObject(object)
  })

  test('setCallHeight', () => {
    const localizedState = { ...InitialUIState() }
    const argument = '12'
    mutations.setCallHeight(localizedState, argument)
    expect(localizedState.callHeight).toEqual(argument)
  })

  test('setCallHeight with number', () => {
    const localizedState = { ...InitialUIState() }
    const argument = 12 // Does not have to be string
    mutations.setCallHeight(localizedState, argument)
    expect(localizedState.callHeight).toEqual(argument)
  })

  test('updateRecentGlyphs', () => {
    const localizedState = { ...InitialUIState() }
    const argument = {
      pack: {
        name: 'Count 1',
        artist: 'Dina Brodsky',
        id: '0903',
        stickerURLs: [
          'https://satellite.mypinata.cloud/ipfs/QmS1NEujgAT8iogdK3jsQzDf751U6LCSpCNojg1JJhs3zz/$1/hawk.webp',
          'https://satellite.mypinata.cloud/ipfs/QmS1NEujgAT8iogdK3jsQzDf751U6LCSpCNojg1JJhs3zz/$1/ducklings.webp',
          'https://satellite.mypinata.cloud/ipfs/QmS1NEujgAT8iogdK3jsQzDf751U6LCSpCNojg1JJhs3zz/$1/owl.webp',
          'https://satellite.mypinata.cloud/ipfs/QmS1NEujgAT8iogdK3jsQzDf751U6LCSpCNojg1JJhs3zz/$1/penguins.webp',
          'https://satellite.mypinata.cloud/ipfs/QmS1NEujgAT8iogdK3jsQzDf751U6LCSpCNojg1JJhs3zz/$1/robin.webp',
          'https://satellite.mypinata.cloud/ipfs/QmS1NEujgAT8iogdK3jsQzDf751U6LCSpCNojg1JJhs3zz/$1/stork.webp',
          'https://satellite.mypinata.cloud/ipfs/QmS1NEujgAT8iogdK3jsQzDf751U6LCSpCNojg1JJhs3zz/$1/yellow_bird.webp',
        ],
      },
      url: 'https://satellite.mypinata.cloud/ipfs/QmS1NEujgAT8iogdK3jsQzDf751U6LCSpCNojg1JJhs3zz/$1/ducklings.webp',
      count: 1,
    }
    mutations.updateRecentGlyphs(localizedState, argument)
    expect(localizedState.recentGlyphs).toEqual([argument])
  })

  test('updateRecentGlyphs with existing glyph', () => {
    const localizedState = { ...InitialUIState() }
    const argument = {
      pack: {
        name: 'Count 1',
        artist: 'Dina Brodsky',
        id: '0903',
        stickerURLs: [
          'https://satellite.mypinata.cloud/ipfs/QmS1NEujgAT8iogdK3jsQzDf751U6LCSpCNojg1JJhs3zz/$1/hawk.webp',
          'https://satellite.mypinata.cloud/ipfs/QmS1NEujgAT8iogdK3jsQzDf751U6LCSpCNojg1JJhs3zz/$1/ducklings.webp',
          'https://satellite.mypinata.cloud/ipfs/QmS1NEujgAT8iogdK3jsQzDf751U6LCSpCNojg1JJhs3zz/$1/owl.webp',
          'https://satellite.mypinata.cloud/ipfs/QmS1NEujgAT8iogdK3jsQzDf751U6LCSpCNojg1JJhs3zz/$1/penguins.webp',
          'https://satellite.mypinata.cloud/ipfs/QmS1NEujgAT8iogdK3jsQzDf751U6LCSpCNojg1JJhs3zz/$1/robin.webp',
          'https://satellite.mypinata.cloud/ipfs/QmS1NEujgAT8iogdK3jsQzDf751U6LCSpCNojg1JJhs3zz/$1/stork.webp',
          'https://satellite.mypinata.cloud/ipfs/QmS1NEujgAT8iogdK3jsQzDf751U6LCSpCNojg1JJhs3zz/$1/yellow_bird.webp',
        ],
      },
      url: 'https://satellite.mypinata.cloud/ipfs/QmS1NEujgAT8iogdK3jsQzDf751U6LCSpCNojg1JJhs3zz/$1/ducklings.webp',
      count: 1,
    }
    mutations.updateRecentGlyphs(localizedState, argument)
    expect(localizedState.recentGlyphs).toEqual([argument])
    mutations.updateRecentGlyphs(localizedState, argument)
    expect(localizedState.recentGlyphs[0]).toMatchObject({ count: 2 })
    // We use [0] because there is only 1 element in the array
  })

  test('updateMostUsedEmoji', () => {
    const localizedState = { ...InitialUIState() }
    const argument = {
      name: 'thumbup',
      emoji: 'üëç',
      count: 1,
    }
    mutations.updateMostUsedEmoji(localizedState, argument)
    expect(localizedState.mostEmojiUsed).toEqual([
      {
        code: argument.name,
        content: argument.emoji,
        count: argument.count,
      },
    ])
  })

  test('updateMostUsedEmoji with existing glyph', () => {
    const localizedState = { ...InitialUIState() }
    const argument = {
      name: 'thumbup',
      emoji: 'üëç',
      count: 1,
    }
    mutations.updateMostUsedEmoji(localizedState, argument)
    expect(localizedState.mostEmojiUsed).toEqual([
      {
        code: argument.name,
        content: argument.emoji,
        count: argument.count,
      },
    ])
    mutations.updateMostUsedEmoji(localizedState, argument)
    expect(localizedState.mostEmojiUsed[0]).toMatchObject({ count: 2 })
    // We use [0] because there is only 1 element in the array
  })
})
