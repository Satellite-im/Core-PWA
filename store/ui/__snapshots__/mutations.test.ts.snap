// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mutations sendMessage with the same lastMessageSender 1`] = `
Object {
  "at": 1645617999076,
  "from": "0xc61b9bb3a7a0767e3179713f3a5c7a9aedce193c",
  "id": 1645617999076,
  "messages": Array [
    Object {
      "at": 1645617999076,
      "id": 1645617999076,
      "payload": "message",
      "reactions": Array [],
      "replies": Array [],
      "type": "text",
    },
    Object {
      "at": 1645617999076,
      "id": 1645617999076,
      "payload": "elio@example.com",
      "reactions": Array [],
      "replies": Array [],
      "type": "text",
    },
  ],
  "to": "0x07ee55aa48bb72dcc6e9d78256648910de513eca",
  "type": "group",
}
`;

exports[`mutations updateMostUsedGlyph with existing emoji 1`] = `
Array [
  Object {
    "count": 1,
    "pack": Object {
      "artist": "Dina Brodsky",
      "description": "Short description can go here. Lorem ipsum.",
      "id": "0903",
      "name": "Birds",
      "stickerURLs": Array [
        "https://satellite.mypinata.cloud/ipfs/QmXrmiCxkyfpsAY18qzVBPv9TpCbTUToowQEsYy1Pm5C9b/$1/hawk.webp",
        "https://satellite.mypinata.cloud/ipfs/QmXrmiCxkyfpsAY18qzVBPv9TpCbTUToowQEsYy1Pm5C9b/$1/ducklings.webp",
        "https://satellite.mypinata.cloud/ipfs/QmXrmiCxkyfpsAY18qzVBPv9TpCbTUToowQEsYy1Pm5C9b/$1/owl.webp",
        "https://satellite.mypinata.cloud/ipfs/QmXrmiCxkyfpsAY18qzVBPv9TpCbTUToowQEsYy1Pm5C9b/$1/penguins.webp",
        "https://satellite.mypinata.cloud/ipfs/QmXrmiCxkyfpsAY18qzVBPv9TpCbTUToowQEsYy1Pm5C9b/$1/robin.webp",
        "https://satellite.mypinata.cloud/ipfs/QmXrmiCxkyfpsAY18qzVBPv9TpCbTUToowQEsYy1Pm5C9b/$1/stork.webp",
        "https://satellite.mypinata.cloud/ipfs/QmXrmiCxkyfpsAY18qzVBPv9TpCbTUToowQEsYy1Pm5C9b/$1/yellow_bird.webp",
      ],
    },
    "url": "https://satellite.mypinata.cloud/ipfs/QmXrmiCxkyfpsAY18qzVBPv9TpCbTUToowQEsYy1Pm5C9b/$1/ducklings.webp",
  },
  Object {
    "count": 1,
    "pack": Object {
      "artist": "John Treanor",
      "description": "Short description can go here. Lorem ipsum.",
      "id": "0123",
      "name": "Astrobunny",
      "stickerURLs": Array [
        "https://satellite.mypinata.cloud/ipfs/QmaBFFeJksvrfEhwbKYkzbgm7bKFWzHhRECqGxgB52LeHg/$1/AHH.gif",
        "https://satellite.mypinata.cloud/ipfs/QmaBFFeJksvrfEhwbKYkzbgm7bKFWzHhRECqGxgB52LeHg/$1/AHHcloseup.gif",
        "https://satellite.mypinata.cloud/ipfs/QmaBFFeJksvrfEhwbKYkzbgm7bKFWzHhRECqGxgB52LeHg/$1/Coy_02.gif",
        "https://satellite.mypinata.cloud/ipfs/QmaBFFeJksvrfEhwbKYkzbgm7bKFWzHhRECqGxgB52LeHg/$1/Cry.gif",
        "https://satellite.mypinata.cloud/ipfs/QmaBFFeJksvrfEhwbKYkzbgm7bKFWzHhRECqGxgB52LeHg/$1/Cry2.gif",
        "https://satellite.mypinata.cloud/ipfs/QmaBFFeJksvrfEhwbKYkzbgm7bKFWzHhRECqGxgB52LeHg/$1/Laugh.webp",
        "https://satellite.mypinata.cloud/ipfs/QmaBFFeJksvrfEhwbKYkzbgm7bKFWzHhRECqGxgB52LeHg/$1/Luv.gif",
        "https://satellite.mypinata.cloud/ipfs/QmaBFFeJksvrfEhwbKYkzbgm7bKFWzHhRECqGxgB52LeHg/$1/Luv_02.gif",
        "https://satellite.mypinata.cloud/ipfs/QmaBFFeJksvrfEhwbKYkzbgm7bKFWzHhRECqGxgB52LeHg/$1/Sad3.gif",
        "https://satellite.mypinata.cloud/ipfs/QmaBFFeJksvrfEhwbKYkzbgm7bKFWzHhRECqGxgB52LeHg/$1/ThumbsDownNew.gif",
        "https://satellite.mypinata.cloud/ipfs/QmaBFFeJksvrfEhwbKYkzbgm7bKFWzHhRECqGxgB52LeHg/$1/ThumbsUP.gif",
        "https://satellite.mypinata.cloud/ipfs/QmaBFFeJksvrfEhwbKYkzbgm7bKFWzHhRECqGxgB52LeHg/$1/ThumbsUPNew.gif",
      ],
    },
    "url": "https://satellite.mypinata.cloud/ipfs/QmaBFFeJksvrfEhwbKYkzbgm7bKFWzHhRECqGxgB52LeHg/$1/Coy_02.gif",
  },
  Object {
    "count": 1,
    "pack": Object {
      "artist": "John Treanor",
      "description": "Short description can go here. Lorem ipsum.",
      "id": "0123",
      "name": "Astrobunny",
      "stickerURLs": Array [
        "https://satellite.mypinata.cloud/ipfs/QmaBFFeJksvrfEhwbKYkzbgm7bKFWzHhRECqGxgB52LeHg/$1/AHH.gif",
        "https://satellite.mypinata.cloud/ipfs/QmaBFFeJksvrfEhwbKYkzbgm7bKFWzHhRECqGxgB52LeHg/$1/AHHcloseup.gif",
        "https://satellite.mypinata.cloud/ipfs/QmaBFFeJksvrfEhwbKYkzbgm7bKFWzHhRECqGxgB52LeHg/$1/Coy_02.gif",
        "https://satellite.mypinata.cloud/ipfs/QmaBFFeJksvrfEhwbKYkzbgm7bKFWzHhRECqGxgB52LeHg/$1/Cry.gif",
        "https://satellite.mypinata.cloud/ipfs/QmaBFFeJksvrfEhwbKYkzbgm7bKFWzHhRECqGxgB52LeHg/$1/Cry2.gif",
        "https://satellite.mypinata.cloud/ipfs/QmaBFFeJksvrfEhwbKYkzbgm7bKFWzHhRECqGxgB52LeHg/$1/Laugh.webp",
        "https://satellite.mypinata.cloud/ipfs/QmaBFFeJksvrfEhwbKYkzbgm7bKFWzHhRECqGxgB52LeHg/$1/Luv.gif",
        "https://satellite.mypinata.cloud/ipfs/QmaBFFeJksvrfEhwbKYkzbgm7bKFWzHhRECqGxgB52LeHg/$1/Luv_02.gif",
        "https://satellite.mypinata.cloud/ipfs/QmaBFFeJksvrfEhwbKYkzbgm7bKFWzHhRECqGxgB52LeHg/$1/Sad3.gif",
        "https://satellite.mypinata.cloud/ipfs/QmaBFFeJksvrfEhwbKYkzbgm7bKFWzHhRECqGxgB52LeHg/$1/ThumbsDownNew.gif",
        "https://satellite.mypinata.cloud/ipfs/QmaBFFeJksvrfEhwbKYkzbgm7bKFWzHhRECqGxgB52LeHg/$1/ThumbsUP.gif",
        "https://satellite.mypinata.cloud/ipfs/QmaBFFeJksvrfEhwbKYkzbgm7bKFWzHhRECqGxgB52LeHg/$1/ThumbsUPNew.gif",
      ],
    },
    "url": "https://satellite.mypinata.cloud/ipfs/QmaBFFeJksvrfEhwbKYkzbgm7bKFWzHhRECqGxgB52LeHg/$1/ThumbsUP.gif",
  },
  Object {
    "count": 1,
    "pack": Object {
      "artist": "Dina Brodsky",
      "description": "Short description can go here. Lorem ipsum.",
      "id": "0903",
      "name": "Birds",
      "stickerURLs": Array [
        "https://satellite.mypinata.cloud/ipfs/QmS1NEujgAT8iogdK3jsQzDf751U6LCSpCNojg1JJhs3zz/$1/hawk.webp",
        "https://satellite.mypinata.cloud/ipfs/QmS1NEujgAT8iogdK3jsQzDf751U6LCSpCNojg1JJhs3zz/$1/ducklings.webp",
        "https://satellite.mypinata.cloud/ipfs/QmS1NEujgAT8iogdK3jsQzDf751U6LCSpCNojg1JJhs3zz/$1/owl.webp",
        "https://satellite.mypinata.cloud/ipfs/QmS1NEujgAT8iogdK3jsQzDf751U6LCSpCNojg1JJhs3zz/$1/penguins.webp",
        "https://satellite.mypinata.cloud/ipfs/QmS1NEujgAT8iogdK3jsQzDf751U6LCSpCNojg1JJhs3zz/$1/robin.webp",
        "https://satellite.mypinata.cloud/ipfs/QmS1NEujgAT8iogdK3jsQzDf751U6LCSpCNojg1JJhs3zz/$1/stork.webp",
        "https://satellite.mypinata.cloud/ipfs/QmS1NEujgAT8iogdK3jsQzDf751U6LCSpCNojg1JJhs3zz/$1/yellow_bird.webp",
      ],
    },
    "url": "https://satellite.mypinata.cloud/ipfs/QmS1NEujgAT8iogdK3jsQzDf751U6LCSpCNojg1JJhs3zz/$1/ducklings.webp",
  },
]
`;

exports[`mutations.default.addReaction 0 1`] = `undefined`;

exports[`mutations.default.addReaction 1 1`] = `undefined`;

exports[`mutations.default.addReaction 2 1`] = `undefined`;

exports[`mutations.default.addReaction 3 1`] = `undefined`;

exports[`mutations.default.addReaction 4 1`] = `undefined`;

exports[`mutations.default.addReaction 5 1`] = `undefined`;

exports[`mutations.default.sendMessage 0 1`] = `undefined`;

exports[`mutations.default.sendMessage 1 1`] = `undefined`;

exports[`mutations.default.sendMessage 2 1`] = `undefined`;

exports[`mutations.default.sendMessage 3 1`] = `undefined`;

exports[`mutations.default.sendMessage 4 1`] = `undefined`;

exports[`mutations.default.sendMessage 5 1`] = `undefined`;
